planning/behavior_path_planner/autoware_behavior_path_planner_common/src/utils/parking_departure/geometric_parallel_parking.cpp

```diff
@@ -627,4 +962,120 @@ void GeometricParallelParking::setTurningRadius(
     common_params.wheel_base + common_params.front_overhang);
 }
 
+std::vector<PathWithLaneId> GeometricParallelParking::generateClothoidalSequence(
+  const double A, const double L, const double theta, const Pose & start_pose,
+  const Pose & end_pose, const double arc_path_interval, const bool is_left_steering,
+  const bool is_forward)
+{
+  std::vector<PathWithLaneId> clothoidal_sequence;
+  const double R = std::pow(A, 2) / L;
+
+  const auto clothoid_path_first =
+    generateClothoidPath(R, L, start_pose, arc_path_interval, is_left_steering, is_forward);
+
+  auto clothoid_path_second =
+    generateClothoidPath(R, L, end_pose, arc_path_interval, is_left_steering, !is_forward);
+  // reverse path points order
+  std::reverse(clothoid_path_second.points.begin(), clothoid_path_second.points.end());
+  // reverse lane_ids. shoulder, lane order
+  for (auto & p : clothoid_path_second.points) {
+    std::reverse(p.lane_ids.begin(), p.lane_ids.end());
+  }
+
+  if (theta != 0) {
```

Comment:
theta is double and may have floating point error
so better not use exact 0
like
```
if (theta < eps)
```